src/hardware/input/mouseif_virtual_machines.cpp

```diff
@@ -62,124 +48,199 @@ union VMwareButtons {
+		is_deactivating  = true;
 		LOG_MSG("MOUSE (PS/2): VMware protocol disabled");
+	}
+
+	if (is_deactivating && was_active) {
+		mouse_shared.active_vmm = virtualbox.is_active || vmware.is_active;
 		MOUSEPS2_UpdateButtonSquish();
 		MOUSE_UpdateGFX();
 	}
-	buttons._data = 0;
-	counter_w     = 0;
+
+	if (protocol == VmmProtocol::VMware) {
+		vmware.buttons._data = 0;
+		vmware.counter_w     = 0;
+	}
 }
 
-void MOUSEVMM_NotifyInputType(const bool new_use_relative,
-                              const bool new_is_input_raw)
+// ***************************************************************************
+// VirtualBox interface implementation
+// ***************************************************************************
+
+bool MOUSE_VirtualBox_IsSupported()
 {
-	use_relative = new_use_relative;
-	is_input_raw = new_is_input_raw;
+	return mouse_config.virtualbox_mouse &&
+	       (mouse_config.model_ps2 != MouseModelPS2::NoMouse);
 }
 
-static void cmd_get_version()
+void MOUSE_VirtualBox_ClientDisconnected()
 {
-	reg_eax = 0; // protocol version
-	reg_ebx = VMWARE_MAGIC;
+	mouse_vmm_deactivate(VmmProtocol::VirtualBox);
 }
 
-static void cmd_abs_pointer_data()
+void MOUSE_VirtualBox_SetPointerAbsolute(const bool is_absolute)
 {
-	reg_eax = buttons._data;
-	reg_ebx = scaled_x;
-	reg_ecx = scaled_y;
-	reg_edx = static_cast<uint32_t>((counter_w >= 0) ? counter_w
-	                                                 : 0x100 + counter_w);
+	if (is_absolute) {
+		mouse_vmm_activate(VmmProtocol::VirtualBox);
+	} else {
+		mouse_vmm_deactivate(VmmProtocol::VirtualBox);
+	}
+}
 
-	counter_w = 0;
+void MOUSE_VirtualBox_SetPointerVisible(const bool is_visible)
+{
+	if (virtualbox.wants_pointer != is_visible) {
+		virtualbox.wants_pointer = is_visible;
+		if (virtualbox.is_active) {
+			mouse_shared.vmm_wants_pointer = is_visible;
+			MOUSE_UpdateGFX();
+		}
+	}
 }
 
-static void cmd_abs_pointer_status()
+void MOUSE_VirtualBox_GetPointerStatus(VirtualBoxPointerStatus& status)
 {
-	reg_eax = updated ? ABS_UPDATED : ABS_NOT_UPDATED;
-	updated = false;
+	status.absolute_x = scaled_x;
+	status.absolute_y = scaled_y;
 }
 
-static void cmd_abs_pointer_command()
+// ***************************************************************************
+// VMware interface implementation
+// ***************************************************************************
+
+bool MOUSE_VMware_IsSupported()
 {
-	switch (static_cast<VMwareAbsPointer>(reg_ebx)) {
-	case VMwareAbsPointer::Enable: break; // can be safely ignored
-	case VMwareAbsPointer::Relative: MOUSEVMM_Deactivate(); break;
-	case VMwareAbsPointer::Absolute: MOUSEVMM_Activate(); break;
-	default:
-		LOG_WARNING("MOUSE (PS/2): unimplemented VMware subcommand 0x%08x",
-		            reg_ebx);
-		break;
-	}
+	return mouse_config.vmware_mouse &&
+	       (mouse_config.model_ps2 != MouseModelPS2::NoMouse);
 }
 
-static uint32_t port_read_vmware(const io_port_t, const io_width_t)
+void MOUSE_VMware_Activate()
```

Comment:
Yeah, just weird... `MOUSE_VmWare_Activate` would be better, or even `MOUSE_Vmware_Activate`

Reply:
Changed capitalization in the source code to `VmWare` (a 'coding rules capitalization'). Comments are still using the 'marketing' capitalization, `VMware` - as this one is used since version 1.0 released in the late 90's.